tests/test_config.py

```diff
@@ -244,6 +244,16 @@ def testOperators(self):
         c2[".a.b"] = 5
         self.assertNotEqual(c1, c2)
 
+    def testMerge(self):
+        c1 = Config({"a": 1, "c": 2})
+        c2 = Config({"a": 3, "b": "b"})
+        c1.merge(c2)
+        self.assertEqual(c1, {"a": 1, "b": "b", "c": 2})
+
```

Comment:
Should this also check that c2 stays its original value (which should also care of making sure c1 and c2 aren't pointing to the same object)?